### DISCUSSION TITLE
How should ignoreTrailingSlash behave?

### DISCUSSION THREAD

**Discussion by conradthegray** [fastify/fastify#3827] [Q&A]
I just want to clarify some things around `ignoreTrailingSlash` flag.

Let's say we have a route like this `/test/:stuff` and a set of paths: `/test/`, `/test/example/`, `/test/example`, `/test`. `ignoreTrailingSlash` flag is set to `true`.

[The documentation](https://www.fastify.io/docs/latest/Reference/Server/#ignoretrailingslash) says that if `ignoreTrailingSlash` is set tot `true`, the server will ignore trailing slashes in all routes.

Based on that, I'd assume the route above will catch all example paths. This is not the case. The last one is not being caught.

Is this an expected behaviour? Shouldn't the server catch in this case all routes?

**Comment by xtx1130** [comment]
Because of fastify is using `find-my-way` for routing, in `find-my-way`, create a route with `ignoreTrailingSlash ` such as:
```js
const findMyWay = FindMyWay({
  ignoreTrailingSlash: true,
})
findMyWay.on('GET', '/test/:stuff', () => {})
```
it equals:
```js
const findMyWay = FindMyWay({})
findMyWay.on('GET', '/test/:stuff', () => {})
findMyWay.on('GET', '/test/:stuff/', () => {})
```

So you can find the definite urls are: `/test/` `/test/example/` `/test/example`

**Comment by jsumners** [comment]
`:stuff` is not defined as an optional parameter -- https://github.com/delvedor/find-my-way#supported-path-formats

Therefore, `/test` is a different route than `/test/:stuff`.

**Reply by conradthegray** [reply]
Thank you for clarifying this. There is a issue https://github.com/fastify/fastify/issues/3331 asking about the same thing and since this is the expected behaviour, I suggest we can close it

**Reply by jsumners** [reply]
Wildcard routes are slightly different from the case you have described here.

**Reply by conradthegray** [reply]
Can you explain the difference?

**Reply by jsumners** [reply]
`/*` is a greedy match. `/test/:stuff` matches much more narrowly.
